Controllers/TimeTable.php

```diff
@@ -54,40 +54,26 @@ public function post(): Response
         if (!AuthService::userIsAtLeast(Roles::$editor)) {
             return $this->template->displayJson(['Error' => 'Not Authorized'], 403);
         }
-        $jsonPayload = json_decode(file_get_contents('php://input'), true);
-        if (isset($jsonPayload['action']) && $jsonPayload['action'] === 'delete') {
-            $timesheetId = $jsonPayload['timesheetId'];
-            if ($timesheetId) {
-                try {
-                    $this->timesheetRepository->deleteTime($timesheetId);
-                    exit(json_encode(['status' => 'success']));
-                } catch (Exception $e) {
-                    exit(json_encode(['status' => 'error', 'error' => $e->getMessage()]));
-                }
+        $redirectUrl = BASE_URL . '/TimeTable/TimeTable';
+        $actionHandler = new TimeTableActionHandler($this->timeTableService, $this->timesheetRepository);
+
+        if (isset($_POST['action'])) {
+            switch ($_POST['action']) {
```

Comment:
As this assigned an return value in each case "$redirectUrl" it could be written as a match statement.